### PR TITLE
Use generic compile_go_fuzzer for golang projects

### PR DESCRIPTION
**Pull request by catenacyber**
Following #3142, and as request in #2187
Here is the generic `compile_go_fuzzer` and the migration of projects to use it

24 projects in this PR
go-dns and gonids were already using this script
golang and kubernetes were specially handcrafted (golang has seed_corpus in its build script, and kubernetes has an automatic way of naming the fuzzer)
20 projects were automated (see dummy python script below)

5 projects to go :
ipfs has a specific build script I did not dare to touch
golang-protobuf, jsonparser, minify, mtail build scripts seem to be hosted in upstream repos

Plus excelize and go-ethereum which have project.yaml but no build.sh

cc @Dor1s
How do you want to test this ?

Script to automate
```
import sys

comp="""function compile_fuzzer {
  path=$1
  function=$2
  fuzzer=$3

  # Compile and instrument all Go files relevant to this fuzz target.
  go-fuzz -func $function -o $fuzzer.a $path

  # Link Go code ($fuzzer.a) with fuzzing engine to produce fuzz target binary.
  $CXX $CXXFLAGS $LIB_FUZZING_ENGINE $fuzzer.a -o $OUT/$fuzzer
}"""

f = open(sys.argv[1])
check = 0
l = f.read()
f.close()
f = open(sys.argv[1], "wb")
if l.find(comp) > 0:
    print "ok", sys.argv[1]
    l=l.replace(comp,"")
elif l.find(comp2) > 0:
    print "ok2", sys.argv[1]
    l=l.replace(comp2,"")
else:
    print "fail", sys.argv[1]
    sys.exit(0)
l=l.replace("compile_fuzzer", "compile_go_fuzzer")
f.write(l)
``` 
with some more variants up to comp5

### CODE REVIEW

File: infra/base-images/base-builder/compile_go_fuzzer [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash -eu
+# Copyright 2016 Google Inc.
```
Comment (Dor1s) [inline infra/base-images/base-builder/compile_go_fuzzer:2]: nit: 2020

**Review by Dor1s — APPROVED** [review]
awesome work!!! We also need to update the docs, but we can do that later. I'm planning to merge this today to test it out in the builder.

File: infra/base-images/base-builder/compile_go_fuzzer (hunk 1 above)
Comment (catenacyber) [reply infra/base-images/base-builder/compile_go_fuzzer]: thanks for fixing this

**Comment by catenacyber** [comment]
> We also need to update the docs, but we can do that later

Do you want me to do this ? Or will you ?

Also, should we change project.yaml files to include coverage builds ?

**Comment by inferno-chromium** [comment]
@catenacyber - for the remaining 5 projects that need upstream changes, can you please file a tracking bug in their repos for this (probably with this cl as example).

**Comment by catenacyber** [comment]
For coredns, it should hopefully get going with #4689 
grpc-gateway seems broken from before adding coverage
go-ethereum has no build.sh and hence has no coverage working

I do not know why golang is failing.
Locally, I can get it with 
`python infra/helper.py coverage --fuzz-target=csv --corpus-dir=$GOPATH/src/github.com/dvyukov/go-fuzz-corpus/csv/corpus golang`

Log from the failing build is
```
Failed to unpack the corpus for csv. This usually means that corpus backup for a particular fuzz target does not exist. If a fuzz target was added in the last 24 hours, please wait one more day. Otherwise, something is wrong with the fuzz target or the infrastructure, and corpus pruning task does not finish successfully.
```
Is this a naming conflict ?

Every other go project seems to have coverage ok now.

So, there indeed remains :
- golang-protobuf
- ipfs
- jsonparser
- minify
- mtail

**Pull request merged**